lib/localparticipant.js

```diff
@@ -311,15 +334,21 @@ class LocalParticipant extends Participant {
    *   name: 'my-cool-room',
    *   audio: true
    * }).then(function(room) {
-   *   // Publish a video MediaStreamTrack with a custom name
-   *   return room.localParticipant.publishTrack(mediaStreamTrack, {
-   *     name: 'camera'
+   *   return navigator.mediaDevices.getUserMedia({
+   *     video: true
+   *   }).then(function(mediaStream) {
+   *     var mediaStreamTrack = mediaStream.getTracks()[0];
+   *     return room.localParticipant.publishTrack(mediaStreamTrack, {
```

Comment:
@makarandp0 If you are specifying a MediaStreamTrack, then we also have to accept `LocalTrackOptions` so that we internally create a `LocalTrack` based on the MediaStreamTrack. So we can't fold `LocalTrackPublishOptions` to `LocalTrackOptions`. How about we do the following:
```js
/**
 * @typedef {LocalTrackOptions} MediaStreamTrackPublishOptions
 * @property {Track.Priority} [priority='medium']
 */
```
Then we can have two arguments for both signatures:
```webidl
Promise<LocalTrackPublication> publishTrack(LocalTrack track, LocalTrackPublishOptions options);
Promise<LocalTrackPublication> publishTrack(MediaStreamTrack track, MediaStreamTrackPublishOptions options);
```
What do you think?

Reply:
like it much better. Thanks!